tmp/lanelet2_extension/include/lanelet2_extension/regulatory_elements/crosswalk.hpp

```diff
@@ -0,0 +1,81 @@
+// Copyright 2023 Tier IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef LANELET2_EXTENSION__REGULATORY_ELEMENTS__CROSSWALK_HPP_
+#define LANELET2_EXTENSION__REGULATORY_ELEMENTS__CROSSWALK_HPP_
+
+// NOLINTBEGIN(readability-identifier-naming)
+
+#include <lanelet2_core/primitives/BasicRegulatoryElements.h>
+#include <lanelet2_core/primitives/Lanelet.h>
+
+#include <memory>
+#include <vector>
+
+namespace lanelet::autoware
+{
+class Crosswalk : public lanelet::RegulatoryElement
+{
+public:
+  using Ptr = std::shared_ptr<Crosswalk>;
+  static constexpr char RuleName[] = "crosswalk";
+
+  static Ptr make(
+    Id id, const AttributeMap & attributes, const Lanelet & crosswalk_lanelet,
+    const Polygon3d & crosswalk_area, const LineStrings3d & stop_line)
+  {
+    return Ptr{new Crosswalk(id, attributes, crosswalk_lanelet, crosswalk_area, stop_line)};
+  }
+
+  /**
+   * @brief get the relevant crosswalk area
+   * @return crosswalk area
+   */
+  [[nodiscard]] ConstPolygons3d crosswalkArea() const;
+  [[nodiscard]] Polygons3d crosswalkArea();
```

Comment:
It's probably better to make the function name plural since it could return multiple areas.

```suggestion
  [[nodiscard]] ConstPolygons3d crosswalkAreas() const;
  [[nodiscard]] Polygons3d crosswalkAreas();
```

Reply:
fixed in https://github.com/autowarefoundation/autoware_common/pull/187/commits/c572b733aaab97f4cc00b8282714deb9be7d4408.